spaghetto/bin/src/Main.purs

```diff
@@ -193,72 +192,50 @@ data Command a
-            , ArgParser.command [ "info" ]
-                "Query the Registry for information about packages and versions"
-                (SpagoCmd <$> globalArgsParser <*> (RegistryInfo <$> registryInfoArgsParser) <* ArgParser.flagHelp)
-            ] <* ArgParser.flagHelp
-    , ArgParser.command [ "ls" ] "List packages or dependencies" do
-        ArgParser.choose "ls-subcommand"
-          [ ArgParser.command [ "packages" ]
-              "List packages available in the local package set"
-              (SpagoCmd <$> globalArgsParser <*> (LsPackages <$> lsPackagesArgsParser) <* ArgParser.flagHelp)
-          , ArgParser.command [ "deps" ]
-              "List dependencies of the project"
-              (SpagoCmd <$> globalArgsParser <*> (LsDeps <$> lsDepsArgsParser) <* ArgParser.flagHelp)
-          ] <* ArgParser.flagHelp
+  hsubparser $ fold
+    [ commonParser "init" (Init <$> initArgsParser) "Initialise a new project"
+    , commonParser "fetch" (Fetch <$> fetchArgsParser) "Downloads all of the project's dependencies"
+    , commonParser "install" (Install <$> installArgsParser) "Compile the project's dependencies"
+    , commonParser "build" (Build <$> buildArgsParser) "Compile the project"
+    , commonParser "run" (Run <$> runArgsParser) "Run the project"
+    , commonParser "test" (Test <$> testArgsParser) "Test the project"
+    , commonParser "bundle" (Bundle <$> bundleArgsParser) "Bundle the project in a single file"
+    , commonParser "sources" (Sources <$> sourcesArgsParser) "List all the source paths (globs) for the dependencies of the project"
+    , commonParser "repl" (Repl <$> replArgsParser) "Start a REPL"
+    , commonParser "publish" (Publish <$> publishArgsParser) "Publish a package"
+    , command "registry"
+        ( info
+            ( hsubparser $ fold
+                [ commonParser "search" (RegistrySearch <$> registrySearchArgsParser) "Search for package names in the Registry"
+                , commonParser "info" (RegistryInfo <$> registryInfoArgsParser) "Query the Registry for information about packages and versions"
+                ]
+            )
+            (progDesc "Commands to interact with the Registry")
+        )
+    , command "ls"
+        ( info
+            ( hsubparser $ fold
+                [ commonParser "packages" (LsPackages <$> lsPackagesArgsParser) "List packages available in the local package set"
+                , commonParser "deps" (LsDeps <$> lsDepsArgsParser) "List dependencies of the project"
+                ]
+            )
+            (progDesc "List packages or dependencies")
+        )
     ]
-    <* ArgParser.flagHelp
-    <* ArgParser.flagInfo [ "--version" ] "Show the current version" BuildInfo.currentSpagoVersion
+
+-- <* hsubparser (help "")
+-- <* infoOption BuildInfo.currentSpagoVersion (?_)
+-- ArgParser.flagInfo [ "--version" ] "Show the current version" BuildInfo.currentSpagoVersion
```

Comment:
I decided to use a command instead of a flag for printing the version.

This command doesn't trigger printing the help message when `prefShowHelpOnError` is set to `true`.

I considered `infoOption`, but this option always fails. So, it prints both a version and a help message.

Reply:
We'll need a flag for the version - this has come up before, see https://github.com/purescript/spago/issues/608

I'd basically like to mirror the behaviour of `purs`:
``` bash
$ purs --version
0.15.9
```
..so it would we great if we can avoid printing the help message

Reply:
Should I remove the `version` command?

Reply:
> I added a --version flag

Looks great!

> Should I remove the version command?

Yes please! We shouldn't add one unless it's necessary.

Reply:
Removed in 87ccdf6d4db3a4725b735d8fe8de689fe4a73822